src/main/java/it/gov/pagopa/ranking/utils/csv/HeaderColumnNameStrategy.java

```diff
@@ -0,0 +1,49 @@
+package it.gov.pagopa.ranking.utils.csv;
+
+import com.opencsv.bean.*;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.io.StringReader;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/** A strategy to build a csv having column ordered as declaration order and columnName as defined using {@link CsvBindByName} annotation*/
+@Slf4j
+public class HeaderColumnNameStrategy<T> extends HeaderColumnNameMappingStrategy<T> {
+    private final boolean useHeader;
+
+    public HeaderColumnNameStrategy(Class<? extends T> clazz, boolean useHeader) {
+        this.useHeader = useHeader;
+        setType(clazz);
+
+        // Build the header line which respects the declaration order
+        String headerLine = Arrays.stream(clazz.getDeclaredFields())
+                .map(field -> Optional.ofNullable(field.getAnnotation(CsvBindByName.class)).map(CsvBindByName::column)
+                        .or(() -> Optional.ofNullable(field.getAnnotation(CsvDate.class)).map(CsvDate::value))
```

Comment:
this way you are writing the value as the columnName!
If CsvDate works, just add the CsvBindByName annotation to the field together with this annotation
Then here, restore the support to CsvCustomBindByName